comit/src/ethereum.rs

```diff
@@ -45,8 +45,8 @@ impl From<Address> for [u8; 20] {
 impl FromStr for Address {
     type Err = FromHexStrError;
 
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        let bytes = hex::decode(s)?;
+    fn from_str(hex: &str) -> Result<Self, Self::Err> {
+        let bytes = hex::decode(hex.trim_start_matches("0x"))?;
```

Comment:
❓ At some point we said that we want to enforce a certain format rather than allowing either `0x` or no prefix. I don't mind that much allowing bother, but what was your motivation to change this?

Reply:
See [`7387a1c` (#2900)](https://github.com/comit-network/comit-rs/pull/2900/commits/7387a1c2c1251a9cb3202942546557cefbefb7b5).